File: DAL/Models/SearchParam.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DAL.Models
{
    public class SearchParam
    {
        public int Page { get; set; }
        public string Route { get; set; }
        public int InvoiceNumber { get; set; }
        public List<string> OrderStatus { get; set; }
        public string CustomerName { get; set; }
        public int BillingInformationId { get; set; }
        public int StoreId { get; set; }
        /// <summary>
        /// Search in multiple table columns
        /// </summary>
        public string SearchText{ get; set; }
        public int YearId { get; set; }
        public int Week { get; set; }
        public bool AllRecords { get; set; }
        //Invoice date range.
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }

        public SearchParam ShallowCopy()
        {
            return (SearchParam)this.MemberwiseClone();
        }
    }
}

File: clientapp/src/app/shared/model/search-param.ts
import * as _ from 'lodash';

export class SearchParam {
    public Page: number;
    public Route: number;
    public InvoiceNumber: number;
    public OrderStatus: Array<string>;
    public CustomerName: string;
    constructor({
        Page,
        Route,
        InvoiceNumber,
        OrderStatus,
        CustomerName }: { Page?: number, Route?: number, InvoiceNumber?: number, OrderStatus?: Array<string>, CustomerName?: string } = {}) {
        this.Page = Page;
        this.Route = Route;
        this.InvoiceNumber = InvoiceNumber;
        this.OrderStatus = OrderStatus;
        this.CustomerName = CustomerName;
    }

    public quryParams() {
        let str = '';
        str += 'Page=' + this.Page;
        str += this.Route > 0 ? '&Route=' + this.Route : '';
        str += this.InvoiceNumber > 0 ? '&InvoiceNumber=' + this.InvoiceNumber : '';
        str += this.CustomerName ? '&CustomerName=' + this.CustomerName : '';

        if (this.OrderStatus && this.OrderStatus.length > 0) {
            _.forEach(this.OrderStatus, status => {
                str += '&OrderStatus=' + status;
            })
        }

        return str;
    }
}